L1Trigger/TrackFindingTracklet/interface/StubStreamData.h

```diff
@@ -0,0 +1,36 @@
+#ifndef L1Trigger_TrackFindingTracklet_interface_StubStreamData_h
+#define L1Trigger_TrackFindingTracklet_interface_StubStreamData_h
+
+#include "L1Trigger/TrackFindingTracklet/interface/L1TStub.h"
+
+#include <string>
+
+// Used to generate bit-accurate stub stream from TrackBuilder output
+// (This class only needed to support stand-alone running of this code).
+
+namespace trklet {
+
+  class L1TStub;
+
+  class StubStreamData {
+  public:
+    StubStreamData() {}
+
+    StubStreamData(int iSeed, const L1TStub& stub, const std::string& residuals)
+        : iSeed_(iSeed), stub_(stub), residuals_(residuals) {}
+
+    ~StubStreamData() = default;
+
+    int iSeed() const { return iSeed_; }          // Seed type
+    bool valid() const { return (iSeed_ >= 0); }  // Valid stub
+    const L1TStub& stub() const { return stub_; }
+    // String containing valid bit + r coordinate + phi residual + r or z residual.
+    const std::string& residuals() const { return residuals_; }
```

Comment:
I find the name residuals misleading, would prefer something like stubword, fwstub, stub, ...

Reply:
and returning a const reference is not cool with cmssw policy, please change that to std::string

Reply:
Why do you believe it's illegal for a function to return a const ref? My recollection is that during earlier PRs, K.Pedro told us functions should return a (const) ref if the function returns a non-trivial class data member. e.g. This happens in this function https://github.com/cms-sw/cmssw/blob/master/SimTracker/TrackTriggerAssociation/interface/TTTrackAssociationMap.h#L56 in response to K.Pedro comments such as this one https://github.com/cms-sw/cmssw/pull/32549#discussion_r548222615 .

Reply:
Kevin told in me in my reviews that the function shall be defined as normal types. He argued that the user of the function has now the choice if he wants a copy or not.

Reply:
No I am messing up something, you are right the return of the const ref is correct.